beets/ui/__init__.py

```diff
@@ -567,31 +691,39 @@ def _colordiff(a, b, highlight='text_highlight',
             b_out.append(b[b_start:b_end])
         elif op == 'insert':
             # Right only.
-            b_out.append(colorize(highlight, b[b_start:b_end]))
+            words = re.split(r"(\s)", b[b_start:b_end])
+            words_colorized = map(add_mapper, words)
+            b_out.append("".join(words_colorized))
         elif op == 'delete':
```

Comment:
I think (didn't test), these constructions could be replaced by the following (simpler, in my opinion) code, which would also remove the need for the somewhat opaque `*_mapper`s:

```python
b_out.append(re.sub(r"\S+", lambda s: colorize(s, highlight_added)))
```

which should replace all matches of the regex (non-space characters in this case) by the result of applying the function in the second argument, cf. https://docs.python.org/3/howto/regex.html#search-and-replace.